xx.md

```diff
@@ -0,0 +1,135 @@
+
+Below is an example tag array for a `kind 32830` email alert (the entire event is not shown because the tags are encrypted and placed in `content`):
+
+```jsonc
+[
+  ["locale","en-US"],
+  ["timezone","-07:00"],
+  ["cron", "0 15 * * 1"],
+  ["email", "email@example.com"],
+  ["claim", "wss://relay.example.com/", "MYACCESSCODE"],
+  ["feed", "[\"intersection\",[\"relay\",\"wss://relay.example.com/\"],[\"scope\",\"network\"]]"],
+  ["description", "My notification"]
+]
+```
+
+## Email Alerts
+
+Email alert requests are `kind 32830` events which specify a requested email digest.
+
+The following additional tags are defined:
+
+- `email` indicates the user's email address
+- `cron` indicates using cron syntax how often the user would like to be notified
+- `handler` (zero or more) is the address of a [NIP 89](./89.md) handler event, for example `["handler", "31990:<pubkey>:<identifier>", "wss://relay.com", "web"]`
+
+## Web Push Alerts
+
+Web push alert requests are `kind 32832` events which can be used to request web push notifications.
+
+The following additional tags are defined:
+
+- `endpoint` indicates a push notification endpoint
+- `p256dh` indicates the p256dh key
+- `auth` indicates the auth key
+
+The push notification payload SHOULD be a JSON object with the following fields:
+
+- `title` - notification title text
+- `body` - notification body text
+- `event` - a [truncated nostr event](#event-truncation)
+- `relays` - a list of nostr relays
+-
+## Android Push Alerts
+
+Android push alert requests are `kind 32833` events which can be used to request Android push notifications.
+
+The following additional tags are defined:
+
+- `device_token` indicates a Firebase Cloud Messaging token
+
+The push notification SHOULD include the following JSON-encoded data:
+
+- `event` - a [truncated nostr event](#event-truncation)
+- `relays` - a list of nostr relays
+
+## iOS Push Alerts
+
+iOS push alert requests are `kind 32834` events which can be used to request iOS push notifications.
+
+The following additional tags are defined:
+
+- `device_token` indicates an APNs token
+- `bundle_identifier` indicates a iOS app bundle identifier
+
+The push notification SHOULD include the following JSON-encoded data:
+
+- `event` - a [truncated nostr event](#event-truncation)
+- `relays` - a list of nostr relays
+
+## Event Truncation
+
+Most push notification standards limit payloads to 4KB, beyond which notifications are not delivered. Since nostr events may be greater than this limit, notifiers SHOULD truncate events when sending push notifications in the following way:
+
+- `content` MAY be omitted
+- `tags` MAY be omitted
+- `sig` SHOULD be removed if either `content` or `tags` are not sent
+
+# Heartbeats
+
+Clients MAY send `kind 23830` heartbeat events to the notifier when the user is online so that notifiers can avoid sending unnecessary notifications. Heartbeats should be one minute apart.
```

Comment:
For the record, I'm very opposed to this or any NIP that requires the client to continuously sign events without the users input

The best outcome we can hope for is signers get smarter and don't bother the user about these kinds of events. but what mostly likely will happen is the user will get annoyed and click the "approve everything" button, which gives the app full access to the users key.
Either way what will have been created is a great way to track users without them being made aware of it

Reply:
@wcat7 @hzrd149 take a look at the new version, I'm using a one-off secret to sign/encrypt heartbeats, which avoids the signer permission.

Reply:
Would it make sense to make the `secret` required instead of optional? if its optional then that technically means there are two separate ways that the server will have to support in order to verify the heartbeat